integration/volume/mount_test.go

```diff
@@ -112,6 +117,102 @@ func TestRunMountVolumeSubdir(t *testing.T) {
 	}
 }
 
+func TestRunMountImage(t *testing.T) {
+	skip.If(t, versions.LessThan(testEnv.DaemonAPIVersion(), "1.45"), "skip test from new feature")
+	skip.If(t, testEnv.UsingSnapshotter(), "snapshotter not supported")
+
+	ctx := setupTest(t)
+	apiClient := testEnv.APIClient()
+
+	for _, tc := range []struct {
+		name         string
+		cmd          []string
+		volumeTarget string
+		createErr    string
+		startErr     string
+		expected     string
+		skipPlatform string
+	}{
+		{name: "image", cmd: []string{"cat", "/image/foo"}, expected: "bar", skipPlatform: "windows:image mounts not supported on Windows"},
+		// Expected is duplicated because the container runs twice
+		{name: "image_remove", cmd: []string{"cat", "/image/foo"}, expected: "barbar", skipPlatform: "windows:image mounts not supported on Windows"},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			if tc.skipPlatform != "" {
+				platform, reason, _ := strings.Cut(tc.skipPlatform, ":")
+				if testEnv.DaemonInfo.OSType == platform {
+					t.Skip(reason)
+				}
+			}
+
+			testImage := setupTestImage(t, ctx, apiClient)
+			if testImage != "" {
+				defer apiClient.ImageRemove(ctx, testImage, image.RemoveOptions{Force: true})
+			}
+
+			cfg := containertypes.Config{
+				Image: "busybox",
+				Cmd:   tc.cmd,
+			}
+
+			hostCfg := containertypes.HostConfig{
+				Mounts: []mount.Mount{
+					{
+						Type:   mount.TypeImage,
+						Source: testImage,
+						Target: "/image",
+					},
+				},
+			}
+
+			startContainer := func(id string) {
+				startErr := apiClient.ContainerStart(ctx, id, containertypes.StartOptions{})
+				if tc.startErr != "" {
+					assert.ErrorContains(t, startErr, tc.startErr)
+					return
+				}
+				assert.NilError(t, startErr)
+			}
+
+			ctrName := strings.ReplaceAll(t.Name(), "/", "_")
+			create, creatErr := apiClient.ContainerCreate(ctx, &cfg, &hostCfg, &network.NetworkingConfig{}, nil, ctrName)
+			id := create.ID
+			if id != "" {
+				defer apiClient.ContainerRemove(ctx, id, containertypes.RemoveOptions{Force: true})
+			}
+
+			if tc.createErr != "" {
+				assert.ErrorContains(t, creatErr, tc.createErr)
+				return
+			}
+			assert.NilError(t, creatErr, "container creation failed")
+			startContainer(id)
+
+			// Test that the container servives a restart when mounted image is removed
+			if tc.name == "image_remove" {
+				stopErr := apiClient.ContainerStop(ctx, id, containertypes.StopOptions{})
+				assert.NilError(t, stopErr)
+
+				_, removeErr := apiClient.ImageRemove(ctx, testImage, image.RemoveOptions{Force: true})
```

Comment:
We should also test if an `ImageRemove` with `Force: false` fails.

Reply:
Yeah, I don't have that functionality in yet. Working on it now!

Reply:
@vvoland feature is now in this PR

https://github.com/moby/moby/pull/48798/files#diff-3eef05a8fc31dcbc2c37b75125916f512d5b5fd74ae6eb6b9cbd35b94a855381
https://github.com/moby/moby/pull/48798/files#diff-a60d56a02728a9de744c7445cb4b12e1a20dd0e99594b63e327ce7ed55fd63e7R194-R197